src/crawlee/_types.py

```diff
@@ -300,6 +300,21 @@ def __call__(
     ) -> Coroutine[None, None, HttpResponse]: ...
 
 
+class UseStateFunction(Protocol):
+    """Type of a function for performing use state.
+
+    Warning:
+    -------
+    This is an experimental feature. The behavior and interface may change in future versions.
+    """
```

Comment:
I believe this should be written like this following the Google style Python docstrings

```suggestion
    """Type of a function for performing use state.

    Warning:
        This is an experimental feature. The behavior and interface may change in future versions.
    """
```